crates/net/eth-wire/src/p2pstream.rs

```diff
@@ -427,6 +402,31 @@ where
             }
         }
 
+        // poll the pinger to determine if we should send a ping
+        match this.pinger.poll_ping(cx) {
+            Poll::Pending => {}
+            Poll::Ready(Ok(PingerEvent::Ping)) => {
+                // encode the ping message
+                let mut ping_bytes = BytesMut::new();
+                P2PMessage::Ping.encode(&mut ping_bytes);
+
+                // check if the buffer is full
+                if this.outgoing_messages.len() >= MAX_P2P_CAPACITY {
+                    return Poll::Ready(Err(P2PStreamError::SendBufferFull))
+                }
```

Comment:
now that I'm reading this again, imo we don't need this error here and should just always push.

But can do that separately from this PR